### PR TITLE
Add tests for the MaskDetectorsIf algorithm

### PR DESCRIPTION
**Pull request by gemmaguest**
This adds missing unit tests for the MaskDetectorsIf algorithm.

**To test:**

Run MaskDetectorsIfTest.

Refs #18215 .

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/CMakeLists.txt [hunk 1]
```diff
@@ -782,6 +782,7 @@ set ( TEST_FILES
 	MCInteractionVolumeTest.h
 	MagFormFactorCorrectionTest.h
 	MaskBinsFromTableTest.h
+	MaskDetectorsIfTest.h
```
Comment (gvardany) [inline Framework/Algorithms/CMakeLists.txt]: Should these not be ordered alphabetically?

File: Framework/Algorithms/test/MaskDetectorsIfTest.h [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+  const MatrixWorkspace_sptr makeFakeWorkspace() {
+    // create the workspace
+    MatrixWorkspace_sptr ws =
+        WorkspaceCreationHelper::create2DWorkspaceWithRectangularInstrument(
+            1, 2, 1);
+
+    // Default y values are all 2.0. Change them so they're different
+    // for each spectrum (this gives us the values 2.0, 2.1, 2.2, ...)
+    for (size_t wi = 0; wi < ws->getNumberHistograms(); wi++) {
```
Comment (gvardany) [inline Framework/Algorithms/test/MaskDetectorsIfTest.h]: use of pre-increment is favourable here: `++wi`

**Review by gvardany — CHANGES_REQUESTED** [review]
I have only few minor comments inline.

File: Framework/Algorithms/test/MaskDetectorsIfTest.h (hunk 2 above)
Comment (gemmaguest) [inline Framework/Algorithms/test/MaskDetectorsIfTest.h]: Thanks for spotting these. Both now fixed.

**Comment by gvardany** [comment]
thanks :shipit:

**Review by gvardany — APPROVED** [review]

**Pull request merged**